core/client/src/notifications.rs

```diff
@@ -147,32 +208,84 @@ impl<Block: BlockT> StorageNotifications<Block> {
 						}
 					}
 				},
+			};
+			if let Some(child_filters) = child_filters {
+				for (c_key, filters) in child_filters.into_iter() {
+					if let Some((listeners, wildcards)) = self.child_listeners.get_mut(&c_key) {
+						if let Some(filters) = filters {
```

Comment:
This part seems identical with the begining of this function, please extract and re-use, it will also reduce the amount of indentation.

I guess it would be:
`self.remove_listeners(filters: Option<_>, &mut listeners, &mut wildcards)`

and for the above code we would just pass `&mut self.listeners, &mut self.wildcards`

Reply:
There is also the possibility to merge both listeners field (will make access to top a bit more costy which can be partly leverage by using a btreemap which may not be a good idea), so probably better to keep it that way and extract common logic.